pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -613,7 +613,7 @@ def __gather_result_across_time_and_optimizers(self, epoch_output):
 
     def log_train_step_metrics(self, batch_output):
         # when metrics should be logged
-        if self.should_update_logs or self.trainer.fast_dev_run:
+        if self.should_update_logs or (int(self.trainer.fast_dev_run) == 1):
```

Comment:
Why is this only done when `== 1`?

Reply:
if `fast_dev_run > 1`, it should follow `log_every_n_steps`, else it will log at every step. https://github.com/PyTorchLightning/pytorch-lightning/pull/4629#pullrequestreview-529414432